File: 2160-minimum-sum-of-four-digit-number-after-splitting-digits/2160-minimum-sum-of-four-digit-number-after-splitting-digits.cpp
class Solution {
public: // greedy
    int minimumSum(int num) {
        vector<int> temp;
        while(num != 0){
            temp.push_back(num%10);
            num /= 10;
        }
        sort(temp.begin(), temp.end());
        return temp[0]*10 + temp[1]*10 + temp[2] + temp[3];    
    }
};

File: 2160-minimum-sum-of-four-digit-number-after-splitting-digits/2160-minimum-sum-of-four-digit-number-after-splitting-digits.js
/**
 * @param {number} num
 * @return {number}
 */
var minimumSum = function(num) {
    const temp = [];
    while(num != 0){
        temp.push(num%10);
        // num /= 10; careful in JS cause it treat like floating point number
        num = Math.floor(num/10);
    }
    temp.sort((a,b)=>{return a-b});
    return temp[0]*10 + temp[1]*10 + temp[2] + temp[3];
};